platform/security/certificate/security-certificate-generator/src/test/java/org/codice/ddf/security/certificate/generator/CertificateSigningRequestTest.java

```diff
@@ -36,33 +53,72 @@
 
     CertificateSigningRequest csr;
 
+    private static KeyPair makeKeyPair() throws Exception {
+        final KeyPairGenerator keyGen = KeyPairGenerator.getInstance("DSA", "SUN");
+        final SecureRandom random = SecureRandom.getInstance("SHA1PRNG", "SUN");
+
+        keyGen.initialize(1024, random);
+        return keyGen.generateKeyPair();
+    }
+
     @Before
     public void setUp() {
         csr = new CertificateSigningRequest();
     }
 
     @Test
     public void testKeys() throws Exception {
-        assertThat("CSR failed to auto-generate RSA keypair", csr.getSubjectPrivateKey(),
+        assertThat("CSR failed to auto-generate RSA keypair",
+                csr.getSubjectPrivateKey(),
                 instanceOf(PrivateKey.class));
-        assertThat("CSR failed to auto-generate RSA keypair", csr.getSubjectPublicKey(),
+        assertThat("CSR failed to auto-generate RSA keypair",
+                csr.getSubjectPublicKey(),
                 instanceOf(PublicKey.class));
         PublicKey pubKey = mock(PublicKey.class);
         PrivateKey privKey = mock(PrivateKey.class);
         KeyPair kp = new KeyPair(pubKey, privKey);
         csr.setSubjectKeyPair(kp);
-        assertThat("Unable to get mock private key", csr.getSubjectPrivateKey(),
+        assertThat("Unable to get mock private key",
+                csr.getSubjectPrivateKey(),
                 sameInstance(privKey));
-        assertThat("Unable to get mock public key", csr.getSubjectPublicKey(),
+        assertThat("Unable to get mock public key",
+                csr.getSubjectPublicKey(),
                 sameInstance(pubKey));
     }
 
     @Test
+    public void testAddSubjectAlternativeNames() {
+        assertThat("CSR should not have any SAN by default",
+                csr.getSubjectAlternativeNames(),
+                emptyCollectionOf(GeneralName.class));
+        csr.addSubjectAlternativeNames("IP:1.2.3.4", "DNS:A");
+        assertThat(csr.getSubjectAlternativeNames(),
+                contains(new GeneralName(GeneralName.iPAddress, "1.2.3.4"),
+                        new GeneralName(GeneralName.dNSName, "A")));
+        csr.addSubjectAlternativeNames("RID:0.2.1.4", "DNS:A");
```

Comment:
✏️ Not a big deal, but we should try to follow the given/when/then structure and keep the `csr.addSubjectAlternativeNames()` together first and the `assertThat()` together after.